### ISSUE TITLE
add new nushell `def --env` `def --wrapped` syntax

### ISSUE THREAD

**Issue by fdncred** [nushell/vscode-nushell-lang#158]
Kubouch introduced some new syntax in [PR10566](https://github.com/nushell/nushell/pull/10566). We need to support this syntax in the extension.

### Old Env Syntax
```nushell
def-env name [] {}
```
### New Env Syntax
```nushell
def --env name [] {}
```
or (I don't think the order matters)
```nushell
def name --env [] {}
```

### Old Wrapped Syntax
```nushell
extern-wrapped name [] {}
```
### New Wrapped Syntax
```nushell
def --wrapped name [] {}
```
or (I don't think the order matters)
```nushell
def name --wrapped [] {}
```

@glcraft Are you interested in helping with this one too?

**Comment by glcraft** [comment]
I have some question: 

* I can handle any flags with `--\w+` or I can handle only these two flags `--(?:env|wrapped)`. I think it's better for the highlighter to handle all flags with the first solution to make colors consistent as you type. What do you prefer ?
* About the order, it is simpler to implement both flags position at once at once, like `(flags)(name)(flags)`, which makes the following valid _for the highlighter_ : `def --env name --env [] {}`. Is it okay for you or do I need to make 2 separates rules. (Remember, this is for the highlighter only, it does not affect validation code in any case)
* Should I need to wrap multi flags in case def evolves in the future to handle multiple flags, like `def --flags1 --flags2 name [] {}` ?

**Comment by glcraft** [comment]
> Right now, there are only env and wrapped. I haven't heard of any plans to add others.
Ok so I will implement those two only. If necessary in the future we can change for some more easily.
> I think one rule is fine for the highlighter, if that's easier. If two rules is easier, do that.
It's easier to make one rule. I will do so 👍 
> I don't understand this question.
Is it planned that we use def with more than one flag at a time.
As you said earlier, there is no plan to other flags. So are `--env` and `--wrapped` usable in a same definition ?
Something like `def --env --wrapped [] {}`

**Comment by fdncred** [comment]
I'd guess that `def --env --wrapped [] {}` is probably a thing. It means that we pass arguments as strings to an external and then we can also change the environment after that, like doing a cd. So, I think we should plan on that happening.

**Comment by fdncred** [comment]
closed by #160
